src/base/bittorrent/session.cpp

```diff
@@ -1347,6 +1361,48 @@ bool Session::addTorrent_impl(AddTorrentData addData, const MagnetUri &magnetUri
     return true;
 }
 
+bool Session::findIncompleteFiles(TorrentInfo &torrentInfo, QString &savePath) const
+{
+    auto findInDir = [](const QString &dirPath, TorrentInfo &torrentInfo) -> bool
+    {
+        bool found = false;
+        if (torrentInfo.filesCount() == 1) {
+            const QString filePath = dirPath + torrentInfo.filePath(0);
+            if (QFile(filePath).exists()) {
+                found = true;
+            }
+            else if (QFile(filePath + QB_EXT).exists()) {
+                found = true;
+                torrentInfo.renameFile(0, torrentInfo.filePath(0) + QB_EXT);
+            }
+        }
+        else {
+            QStringList allFiles =
+                    findAllFiles(dirPath + Utils::String::fromStdString(torrentInfo.nativeInfo()->files().name()));
+            for (int i = 0; i < torrentInfo.filesCount(); ++i) {
```

Comment:
>The loop does not exit after `found = true;` which looks like an intention to test for a complete match (all the files).

That's it. 
We have to process all files to be able to start with a partially downloaded torrent, when the "Append .!qB extension" option is enabled. In this case we can have this directory content (for example):
```
torrent name/
    file1.ext
    file2.ext.!qB
    file3.ext.!qB
    ...
    fileN.ext
```